bundles/model/org.eclipse.smarthome.model.thing/src/org/eclipse/smarthome/model/thing/internal/GenericThingProvider.xtend

```diff
@@ -448,11 +448,10 @@ class GenericThingProvider extends AbstractProvider<Thing> implements ThingProvi
                             ]
                             val oldThing = thingsMap.get(modelName).findFirst[it.UID == newThing.UID]
                             if (oldThing != null) {
-                                thingsMap.get(modelName).remove(oldThing)
-                                thingsMap.get(modelName).add(newThing)
+                                oldThing.merge(newThing)
                                 logger.debug("Refreshing thing '{}' after successful retry", newThing.UID)
                                 if (!ThingHelper.equals(oldThing, newThing)) {
-                                    notifyListenersAboutUpdatedElement(oldThing, newThing)
+                                    notifyListenersAboutUpdatedElement(oldThing, oldThing)
```

Comment:
LGTM.

The only thing I don't know is, if someone relies on the fact that the old element and the new element are different references.